packages/nuxt/src/core/runtime/nitro/handlers/error.ts

```diff
@@ -1,54 +1,44 @@
 import { joinURL, withQuery } from 'ufo'
 import type { NitroErrorHandler } from 'nitropack'
-import type { H3Error } from 'h3'
-import { getRequestHeaders, send, setResponseHeader, setResponseStatus } from 'h3'
-
-import type { NuxtPayload } from 'nuxt/app'
+import { getRequestHeaders, send, setResponseHeader, setResponseHeaders, setResponseStatus } from 'h3'
 
+import { isJsonRequest } from '../utils/error'
 import { useRuntimeConfig } from '#internal/nitro'
 import { useNitroApp } from '#internal/nitro/app'
-import { isJsonRequest, normalizeError } from '#internal/nitro/utils'
-
-export default <NitroErrorHandler> async function errorhandler (error: H3Error, event) {
-  // Parse and normalize error
-  const { stack, statusCode, statusMessage, message } = normalizeError(error)
+import type { NuxtPayload } from '#app/nuxt'
 
-  // Create an error object
-  const errorObject = {
-    url: event.path,
-    statusCode,
-    statusMessage,
-    message,
-    stack: import.meta.dev && statusCode !== 404
-      ? `<pre>${stack.map(i => `<span class="stack${i.internal ? ' internal' : ''}">${i.text}</span>`).join('\n')}</pre>`
-      : '',
-    // TODO: check and validate error.data for serialisation into query
-    data: error.data as any,
-  } satisfies Partial<NuxtPayload['error']> & { url: string }
-
-  // Console output
-  if (error.unhandled || error.fatal) {
-    const tags = [
-      '[nuxt]',
-      '[request error]',
-      error.unhandled && '[unhandled]',
-      error.fatal && '[fatal]',
-      Number(errorObject.statusCode) !== 200 && `[${errorObject.statusCode}]`,
-    ].filter(Boolean).join(' ')
-    console.error(tags, (error.message || error.toString() || 'internal server error') + '\n' + stack.map(l => '  ' + l.text).join('  \n'))
+export default <NitroErrorHandler> async function errorhandler (error, event, { defaultHandler }) {
+  if (isJsonRequest(event)) {
+    // let Nitro handle JSON errors
+    return
   }
+  // invoke default Nitro error handler (which will log appropriately if required)
+  const defaultError = await defaultHandler(error, event, { json: true })
 
   if (event.handled) { return }
```

Comment:
defaultHandler does not handle response but only return one, it might be a misleading leftover check(also I suggest you rename `defaultError` to `response` or *response)